controllers/gslb_controller_reconciliation.go

```diff
@@ -190,7 +222,7 @@ func (r *GslbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		Msg("Resolved LoadBalancer and Server configuration referenced by Ingress")
 
 	// == external-dns dnsendpoints CRs ==
-	dnsEndpoint, err := r.gslbDNSEndpoint(gslb)
+	dnsEndpoint, err := r.gslbDNSEndpoint(gslb, zone)
```

Comment:
A GSLB may have servers with different zones.
What about storing the zone in the status of the GSLB, perhaps by extending Adapting the `Server` datastructure?

Reply:
Good idea, will add.